compiler-rt/lib/scudo/standalone/flags_parser.cpp

```diff
@@ -136,8 +143,17 @@ bool FlagParser::runHandler(const char *Name, const char *Value) {
       break;
     case FlagType::FT_int:
       char *ValueEnd;
-      *reinterpret_cast<int *>(Flags[I].Var) =
-          static_cast<int>(strtol(Value, &ValueEnd, 10));
+      long V = strtol(Value, &ValueEnd, 10);
+      static_assert(INT_MAX <= LONG_MAX);
```

Comment:
I don't think this can ever be the case? `sizeof(long) >= sizeof(int)`. Can just remove.

Reply:
That's the point, a `static_assert` should be something that should always be the case. Can remove, but the point was to explicitly state the assumption of the code below

Reply:
But doesn't the standard say this? It would have to be a non-standards compliant compiler where this failed.

It doesn't really hurt to have this though.

Reply:
Yes, the standard says this, but this is machine-checked documentation that this is the case.